cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -12,361 +12,329 @@
-		dbName    = mustGetenv("DB_NAME") // e.g. 'my-database'
-	)
-
-	dbURI := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s?parseTime=true", dbUser, dbPwd, dbTCPHost, dbPort, dbName)
-
-	// [START_EXCLUDE]
-	// [START cloud_sql_mysql_databasesql_sslcerts]
-	// (OPTIONAL) Configure SSL certificates
-	// For deployments that connect directly to a Cloud SQL instance without
-	// using the Cloud SQL Proxy, configuring SSL certificates will ensure the
-	// connection is encrypted. This step is entirely OPTIONAL.
-	dbRootCert := os.Getenv("DB_ROOT_CERT") // e.g., '/path/to/my/server-ca.pem'
-	if dbRootCert != "" {
-		var (
-			dbCert = mustGetenv("DB_CERT") // e.g. '/path/to/my/client-cert.pem'
-			dbKey  = mustGetenv("DB_KEY")  // e.g. '/path/to/my/client-key.pem'
-		)
-		pool := x509.NewCertPool()
-		pem, err := ioutil.ReadFile(dbRootCert)
-		if err != nil {
-			return nil, err
-		}
-		if ok := pool.AppendCertsFromPEM(pem); !ok {
-			return nil, err
-		}
-		cert, err := tls.LoadX509KeyPair(dbCert, dbKey)
-		if err != nil {
-			return nil, err
-		}
-		mysql.RegisterTLSConfig("cloudsql", &tls.Config{
-			RootCAs:               pool,
-			Certificates:          []tls.Certificate{cert},
-			InsecureSkipVerify:    true,
-			VerifyPeerCertificate: verifyPeerCertFunc(pool),
-		})
-		dbURI += "&tls=cloudsql"
-	}
-	// [END cloud_sql_mysql_databasesql_sslcerts]
-	// [END_EXCLUDE]
-
-	// dbPool is the pool of database connections.
-	dbPool, err := sql.Open("mysql", dbURI)
-	if err != nil {
-		return nil, fmt.Errorf("sql.Open: %v", err)
-	}
-
-	// [START_EXCLUDE]
-	configureConnectionPool(dbPool)
-	// [END_EXCLUDE]
-
-	return dbPool, nil
-	// [END cloud_sql_mysql_databasesql_create_tcp]
-}
-
-// verifyPeerCertFunc returns a function that verifies the peer certificate is
-// in the cert pool.
-func verifyPeerCertFunc(pool *x509.CertPool) func([][]byte, [][]*x509.Certificate) error {
-	return func(rawCerts [][]byte, _ [][]*x509.Certificate) error {
-		if len(rawCerts) == 0 {
-			return errors.New("no certificates available to verify")
-		}
-
-		cert, err := x509.ParseCertificate(rawCerts[0])
-		if err != nil {
-			return err
-		}
-
-		opts := x509.VerifyOptions{Roots: pool}
-		if _, err = cert.Verify(opts); err != nil {
-			return err
-		}
-		return nil
-	}
-}
-
-// configureConnectionPool sets database connection pool properties.
-// For more information, see https://golang.org/pkg/database/sql
-func configureConnectionPool(dbPool *sql.DB) {
-	// [START cloud_sql_mysql_databasesql_limit]
-
-	// Set maximum number of connections in idle connection pool.
-	dbPool.SetMaxIdleConns(5)
-
-	// Set maximum number of open connections to the database.
-	dbPool.SetMaxOpenConns(7)
-
-	// [END cloud_sql_mysql_databasesql_limit]
-
-	// [START cloud_sql_mysql_databasesql_lifetime]
-
-	// Set Maximum time (in seconds) that a connection can remain open.
-	dbPool.SetConnMaxLifetime(1800 * time.Second)
-
-	// [END cloud_sql_mysql_databasesql_lifetime]
-}
+var indexHTML = `
```

Comment:
A separate file is much nicer, but getting it to work in Cloud Functions requires the code look in a special place to find the template. So rather than introduce runtime-specific concerns into the app, I chose to imbed the template for simplicity.